core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -138,22 +212,19 @@ public UnassignedInfo getUnassignedInfo() {
         return this.unassignedInfo;
     }
 
-    /** Return a map of node to decision for shard allocation */
-    public Map<DiscoveryNode, Decision> getNodeDecisions() {
-        return this.nodeToDecision;
+    /** Return the remaining allocation delay for this shard in nanoseconds */
```

Comment:
as rest api displays this as millis, we should also return millis here. (Best would be to directly construct ClusterAllocationExplanation with remainingDelayMillis.)